code/common/pcapdecoder.q

```diff
@@ -0,0 +1,87 @@
+\d .pcap
+
+// size of headers in bytes and dict of protocol code conversions
+globheader: 24;
+packetheader: 16;
+allcodes:(enlist 6)!(enlist `TCP);
+
+
+buildtable:{[file]
+ // initial x and byte cut points removed from array list to make table
+ // gettablerow is iterated over each datapacket, extracting data
+
+ data:1_ last each {[n]
+  filebytesize: count n;
+  gettablerow[n;]\[{y>(first x[0])+40}[;filebytesize];(),0]    
+  } read1 file;
+ 
+ data: update sym:` from data;
+ `time`sym xcols data
+ }
+
+
+gettablerow:{[n;x]  // data for a single row
+ time:     first gettime[n;x];
+ flags:    getflags[n;x];
+ protocol: getprotocol[n;x];
+ 
+ totallength: (enlist 2;enlist "h")1: datafromfile[n;x;18;2];
```

Comment:
I might be misunderstanding this line, but I think what's happening here is reading two bytes as a short - I think this might be simplified with `sv` when you're reading a single value e.g.

```q
q)(enlist 2;enlist"h")1:0x0010
16
q)type (enlist 2;enlist"h")1:0x0010
0h
q)first (enlist 2;enlist"h")1:0x0010
,16h
q)first first (enlist 2;enlist"h")1:0x0010
16h
q)0x0 sv 0x0010
16h
q)type 0x0 sv 0x0010
-5h
```

with `1:` you end up with a nested list of shorts, but with `sv` you get an atom, which probably simplifies things later on as well

https://code.kx.com/q/ref/sv/#bytes-to-integer